src/compiler/tsc.ts

```diff
@@ -180,11 +180,26 @@ namespace ts {
                 reportDiagnostic(createCompilerDiagnostic(Diagnostics.The_current_host_does_not_support_the_0_option, "--project"));
                 return sys.exit(ExitStatus.DiagnosticsPresent_OutputsSkipped);
             }
-            configFileName = normalizePath(combinePaths(commandLine.options.project, "tsconfig.json"));
             if (commandLine.fileNames.length !== 0) {
                 reportDiagnostic(createCompilerDiagnostic(Diagnostics.Option_project_cannot_be_mixed_with_source_files_on_a_command_line));
                 return sys.exit(ExitStatus.DiagnosticsPresent_OutputsSkipped);
             }
+
+            let fileOrDirectory = normalizePath(commandLine.options.project);
+            if (!fileOrDirectory /* current directory */ || sys.directoryExists(fileOrDirectory)) {

```

Comment:
This is because `normalizePath(".")` gives empty string. `sys.directoryExists` function returns `false` when the input string is empty so I added it.